File: src/frontend/src/app/components/financeiro/plano-contas/model/plano-contas-dto.ts
export class PlanoContasDTO {
  id?: string;
  codigo: string;
  descricao: string;
  tipo: string; // RECEITA, DESPESA, ATIVO, PASSIVO
  planoPaiId?: string;
  planoPaiDescricao?: string;
  unidadeNegocioId?: string;
  unidadeNegocioNome?: string;
  ativo?: boolean;
  analitico?: boolean;
  nivel?: number;
  createdAt?: Date;
  updatedAt?: Date;
  createdBy?: string;
  updatedBy?: string;

  constructor(
    id?: string,
    codigo = '',
    descricao = '',
    tipo = '',
    planoPaiId?: string,
    planoPaiDescricao?: string,
    unidadeNegocioId?: string,
    unidadeNegocioNome?: string,
    ativo?: boolean,
    analitico?: boolean,
    nivel?: number,
    createdAt?: Date,
    updatedAt?: Date,
    createdBy?: string,
    updatedBy?: string
  ) {
    this.id = id;
    this.codigo = codigo;
    this.descricao = descricao;
    this.tipo = tipo;
    this.planoPaiId = planoPaiId;
    this.planoPaiDescricao = planoPaiDescricao;
    this.unidadeNegocioId = unidadeNegocioId;
    this.unidadeNegocioNome = unidadeNegocioNome;
    this.ativo = ativo;
    this.analitico = analitico;
    this.nivel = nivel;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/planocontas/PlanoContasDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.planocontas;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class PlanoContasDTO implements DTO {

    private UUID id;
    private String codigo;
    private String descricao;
    private String tipo; // RECEITA, DESPESA, ATIVO, PASSIVO
    private UUID planoPaiId;
    private String planoPaiDescricao; // Para exibição
    private UUID unidadeNegocioId;
    private String unidadeNegocioNome; // Para exibição
    private Boolean ativo;
    private Boolean analitico; // Calculado
    private Integer nivel; // Calculado
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
